sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -10,109 +10,75 @@
-
-                if (challenge != null)
+                if (async)
                 {
-                    // update the cached challenge if not yet set or different from the current challenge (e.g. moved tenants)
-                    if (_challenge == null || !challenge.Equals(_challenge))
-                    {
-                        _challenge = challenge;
-                    }
-
-                    // authenticate the request and resend
-                    await AuthenticateRequestAsync(message, async, challenge).ConfigureAwait(false);
-
-                    if (async)
-                    {
-                        await ProcessNextAsync(message, pipeline).ConfigureAwait(false);
-                    }
-                    else
-                    {
-                        ProcessNext(message, pipeline);
-                    }
+                    await ProcessNextAsync(message, pipeline).ConfigureAwait(false);
+                }
+                else
+                {
+                    ProcessNext(message, pipeline);
                 }
+
+                // set the content to the original content.
+                message.Request.Content = originalContent;
             }
         }
 
-        private async Task AuthenticateRequestAsync(HttpMessage message, bool async, AuthenticationChallenge challenge)
+        protected override bool TryGetTokenRequestContextFromChallenge(HttpMessage message, out TokenRequestContext context)
         {
-            if (_headerValue is null || DateTimeOffset.UtcNow >= _refreshOn)
+            // if this policy doesn't have _challenge cached try to get it from the static challenge cache
+            AuthenticationChallenge challenge = AuthenticationChallenge.GetChallenge(message);
+
+            if (challenge == null)
             {
-                AccessToken token = async ?
-                        await _credential.GetTokenAsync(new TokenRequestContext(challenge.Scopes, message.Request.ClientRequestId), message.CancellationToken).ConfigureAwait(false) :
-                        _credential.GetToken(new TokenRequestContext(challenge.Scopes, message.Request.ClientRequestId), message.CancellationToken);
+                return false;
+            }
 
-                _headerValue = BearerChallengePrefix + token.Token;
-                _refreshOn = token.ExpiresOn - TimeSpan.FromMinutes(2);
+            // update the cached challenge if not yet set or different from the current challenge (e.g. moved tenants)
+            if (_challenge == null || !challenge.Equals(_challenge))
+            {
+                _challenge = challenge;
             }
 
-            message.Request.Headers.SetValue(HttpHeader.Names.Authorization, _headerValue);
+            context = new TokenRequestContext(_challenge.Scopes, message.Request.ClientRequestId);
+            return true;
         }
 
         internal class AuthenticationChallenge
```

Comment:
I thought you were going to use the caching of the base class.

Reply:
We use the token caching which also handles background refreshes, but not the scopes. I suppose all caching could evolve to offload to the base at some point. But for this initial refactoring, I wanted to change as little as possible.